crates/uv/tests/sync.rs

```diff
@@ -205,3 +205,89 @@ fn empty() -> Result<()> {
 
     Ok(())
 }
+
+#[test]
+fn mixed_requires_python() -> Result<()> {
+    let context = TestContext::new_with_versions(&["3.8", "3.12"]);
+
+    // Create a workspace root with a minimum Python requirement of Python 3.12.
+    let pyproject_toml = context.temp_dir.child("pyproject.toml");
+    pyproject_toml.write_str(
+        r#"
+        [project]
+        name = "albatross"
+        version = "0.1.0"
+        requires-python = ">=3.12"
+        dependencies = ["bird-feeder", "tqdm>=4,<5"]
+
+        [tool.uv.sources]
+        bird-feeder = { workspace = true }
+
+        [tool.uv.workspace]
+        members = ["packages/*"]
+
+        [build-system]
+        requires = ["hatchling"]
+        build-backend = "hatchling.build"
+        "#,
+    )?;
+
+    let src = context.temp_dir.child("src").child("albatross");
+    src.create_dir_all()?;
+
+    let init = src.child("__init__.py");
+    init.touch()?;
+
+    // Create a child with a minimum Python requirement of Python 3.8.
+    let child = context.temp_dir.child("packages").child("bird-feeder");
+    child.create_dir_all()?;
+
+    let src = context.temp_dir.child("src").child("bird_feeder");
+    src.create_dir_all()?;
+
+    let init = src.child("__init__.py");
+    init.touch()?;
+
+    let pyproject_toml = child.child("pyproject.toml");
+    pyproject_toml.write_str(
+        r#"
+        [project]
+        name = "bird-feeder"
+        version = "0.1.0"
+        requires-python = ">=3.8"
+        "#,
+    )?;
+
+    // Running `uv sync` should succeed, locking for Python 3.12.
+    uv_snapshot!(context.filters(), context.sync().arg("-p").arg("3.12"), @r###"
+    success: true
+    exit_code: 0
+    ----- stdout -----
+
+    ----- stderr -----
+    warning: `uv sync` is experimental and may change without warning
+    Using Python 3.12.[X] interpreter at: [PYTHON-3.12]
+    Creating virtualenv at: .venv
+    warning: `uv.sources` is experimental and may change without warning
+    Resolved 4 packages in [TIME]
+    Prepared 3 packages in [TIME]
+    Installed 3 packages in [TIME]
+     + albatross==0.1.0 (from file://[TEMP_DIR]/)
+     + bird-feeder==0.1.0 (from file://[TEMP_DIR]/packages/bird-feeder)
+     + tqdm==4.66.2
+    "###);
+
+    // Running `uv sync` again should succeed.
+    uv_snapshot!(context.filters(), context.sync().arg("-p").arg("3.8"), @r###"
+    success: false
+    exit_code: 2
+    ----- stdout -----
+
+    ----- stderr -----
+    warning: `uv sync` is experimental and may change without warning
+    Using Python 3.8.[X] interpreter at: [PYTHON-3.8]
+    error: The requested Python interpreter (3.8.[X]) is incompatible with the project Python requirement: `>=3.12`
```

Comment:
We should add a specific error message for this case that tells you how to run something on 3.8 with only bird-feeder.

Reply:
@konstin  -- What's the recommended workflow? Can we even support it right now?

Reply:
I'd say we check if the request version is in the union, but not in the intersection, and if so, recommend `uv pip install -p 3.8 -e . <older-python-version-project>`, which _should_ work, i can test this after https://github.com/astral-sh/uv/pull/5644#issuecomment-2260706514

Reply:
Yes, works.

Reply:
Now i checked the right way and `cargo run venv -p 3.8 && cargo run pip install -e packages/foo_lib/` indeed works.

Reply:
Will do separately.